src/plugins/intel_gpu/src/kernel_selector/kernels/lora/lora_kernel_ref.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lora_kernel_ref.h"
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+ParamsKey LoRAKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+
+    k.EnableInputLayout(DataLayout::bfyx);
+    k.EnableOutputLayout(DataLayout::bfyx);
+
+    k.EnableDifferentTypes();
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    k.EnableDynamicShapesSupport();
+
+    return k;
+}
+
+CommonDispatchData LoRAKernelRef::SetDefault(const lora_params& params) const {
+    CommonDispatchData dispatchData;
+
+    const auto& output = params.outputs[0];
+
+    const auto& lora_rank_dim = params.inputs[3].Feature();
+    size_t lora_rank = lora_rank_dim.is_dynamic ? 1 : lora_rank_dim.v;
+
+    dispatchData.gws = { output.Batch().v * output.Feature().v,
+                         Align(output.Y().v * output.X().v, lora_rank),
+                         1};
+
+    dispatchData.lws = { 1, lora_rank, 1 };
+
+    return dispatchData;
+}
+
+JitConstants LoRAKernelRef::GetJitConstants(const lora_params& params) const {
+    auto jit = Parent::GetJitConstants(params);
+
+    jit.AddConstant(MakeJitConstant("MAX_LORA_RANK", 256));
```

Comment:
Can we simply use a tmp buffer with LORA_RANK size? I see "MAX_LORA_RANK" is only used as tmp buffer in ref kernel, and each LWS is lora_rank.

Reply:
Variable length arrays are not supported in OpenCL

Reply:
I see LORA_RANK  is a jit const. Are you going to put it in shape_info as dynamic?

Reply:
LORA_RANK is dynamic and this jit constant is expanded as `shape_info[some_idx]` on kernel